pkg/helm/config.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"fmt"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/kube"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/kubernetes-sigs/minibroker/pkg/log"
+)
+
+//go:generate mockgen -destination=./mocks/mock_config.go -package=mocks github.com/kubernetes-sigs/minibroker/pkg/helm ConfigProvider,ConfigInitializer,ConfigInitializerProvider
+
+const (
+	driver = "secret"
+
+	// Empty values for these mean the internal defaults will be used.
+	defaultKubeConfig = ""
+	defaultContext    = ""
+)
+
+// ConfigProvider is the interface that wraps the Provide method for the upstream Helm configuration
+// provider.
+type ConfigProvider interface {
+	Provide(namespace string) (interface{}, error)
```

Comment:
I can probably use the right type instead of `interface{}` here. This was needed at one point in my experiment and I forgot to make it right.

Reply:
Actually, not. Since the `action.Config` has an `Init` method that gets called from within the function, we need to return the interface. Even though this is meh, it works as expected and worth keeping it this way.